src/app/locator/qgsinbuiltlocatorfilters.cpp

```diff
@@ -702,3 +708,211 @@ void QgsBookmarkLocatorFilter::triggerResult( const QgsLocatorResult &result )
+          result.userData = data;
+          result.score = 1.0;
+          emit resultFetched( result );
+        }
+      }
+      else
+      {
+        if ( withinWgs84 )
+        {
+          QgsLocatorResult result;
+          result.filter = this;
+          result.displayString = tr( "Go to %1 %2 (%3)" ).arg( QString::number( point.x() ), QString::number( point.y() ), wgs84Crs.authid() );
+          result.userData = data;
+          result.score = 1.0;
+          emit resultFetched( result );
+        }
+      }
+    }
+    return;
+  }
+
+  QMap<int, double> scales;
+  scales[0] = 739571909;
+  scales[1] = 369785954;
+  scales[2] = 184892977;
+  scales[3] = 92446488;
+  scales[4] = 46223244;
+  scales[5] = 23111622;
+  scales[6] = 11555811;
+  scales[7] = 5777905;
+  scales[8] = 2888952;
+  scales[9] = 1444476;
+  scales[10] = 722238;
+  scales[11] = 361119;
+  scales[12] = 180559;
+  scales[13] = 90279;
+  scales[14] = 45139;
+  scales[15] = 22569;
+  scales[16] = 11284;
+  scales[17] = 5642;
+  scales[18] = 2821;
+  scales[19] = 1500;
+  scales[20] = 1000;
+
+  QUrl url( string );
+  if ( url.isValid() )
+  {
+    double scale = 0.0;
+    bool okX = false;
+    bool okY = false;
+    double posX = 0.0;
+    double posY = 0.0;
+    if ( url.hasFragment() )
+    {
+      // Check for OSM/Leaflet/OpenLayers pattern (e.g. http://www.openstreetmap.org/#map=6/46.423/4.746)
+      QStringList fragments = url.fragment().split( '&' );
+      for ( const QString &fragment : fragments )
+      {
+        if ( fragment.startsWith( QStringLiteral( "map=" ) ) )
+        {
+          QStringList params = fragment.mid( 4 ).split( '/' );
+          if ( params.size() >= 3 )
+          {
+            if ( scales.contains( params.at( 0 ).toInt() ) )
+            {
+              scale = scales.value( params.at( 0 ).toInt() );
+            }
+            posX = params.at( 2 ).toDouble( &okX );
+            posY = params.at( 1 ).toDouble( &okY );
+          }
+          break;
+        }
+      }
+    }
+
+    if ( !okX && !okY && string.indexOf( QStringLiteral( "google.com/maps/" ) ) != -1 )
+    {
+      QRegularExpression locationRx( QStringLiteral( "\\/@([0-9\\-\\.\\,]*)z" ) );
+      QRegularExpressionMatch match = locationRx.match( string );
+      if ( match.hasMatch() )
+      {
+        QStringList params = match.captured( 1 ).split( ',' );
+        if ( params.size() == 3 )
+        {
+          if ( scales.contains( params.at( 2 ).toInt() ) )
+          {
+            scale = scales.value( params.at( 2 ).toInt() );
+          }
+          posX = params.at( 1 ).toDouble( &okX );
+          posY = params.at( 0 ).toDouble( &okY );
+        }
+      }
+    }
+
+    if ( okX && okY )
+    {
+      QVariantMap data;
+      if ( scale > 0.0 )
+      {
+        data.insert( QStringLiteral( "scale" ), scale );
+      }
+
+      QgsPointXY point( posX, posY );
+      bool withinWgs84 = wgs84Crs.bounds().contains( point );
```

Comment:
won't these always be wgs84 ?

Reply:
Yes, except when someone passes on a flawed URL with erroneous lon and/or lat, which then leads to a transformation error. I'd rather deal with those via a bounds check than an error catch.